### PR TITLE
feat: patch to handle replayer errors

### PR DESCRIPTION
**Pull request by pauldambra**
stacked on top of https://github.com/PostHog/posthog/pull/19861

using pnpm patch to force an error handler into rrweb

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 1]
```diff
@@ -179,7 +179,8 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
         setIsFullScreen: (isFullScreen: boolean) => ({ isFullScreen }),
         skipPlayerForward: (rrWebPlayerTime: number, skip: number) => ({ rrWebPlayerTime, skip }),
         incrementClickCount: true,
-        playerErrorSeen: (errorEvent: ErrorEvent) => ({ errorEvent }),
+        // it should be better typed than this but needs teseting
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:182]:
```suggestion
        // it should be better typed than this but we have no control what is thrown into here
```

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 2]
```diff
@@ -467,15 +468,13 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
-                errorEvent.message + errorEvent.filename + errorEvent.lineno + errorEvent.colno
-            )
+        playerErrorSeen: ({ error }) => {
+            const fingerprint = encodeURIComponent(error.message + error.filename + error.lineno + error.colno)
             if (values.reportedReplayerErrors.has(fingerprint)) {
                 return
             }
             const extra = { fingerprint, recordingId: values.sessionRecordingId }
-            captureException(errorEvent.error, {
+            captureException(error.error, {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:477]:
```suggestion
            captureException(error, {
```

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 3]
```diff
@@ -533,15 +532,20 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
+            const config = {
                 root: values.rootFrame,
                 ...COMMON_REPLAYER_CONFIG,
                 // these two settings are attempts to improve performance of running two Replayers at once
                 // the main player and a preview player
                 mouseTail: props.mode !== SessionRecordingPlayerMode.Preview,
                 useVirtualDom: false,
                 plugins,
-            })
+                // ts-expect-error - this is a valid option added by patching rrweb player 🙈
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:543]:
```suggestion
                // this is an option added by patching rrweb player 🙈
```

**Review by daibhin — APPROVED** [review]

**Comment by pauldambra** [comment]
I'm actually scared to rebase this change 🙈 

going to open a fresh PR

**Pull request closed without merging**